plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1010,43 +1010,47 @@ void producer_plugin::plugin_initialize(const boost::program_options::variables_
       if (chain.is_eos_vm_oc_enabled()) {
          // EOS VM OC requires 4.2TB Virtual for each executing thread. Make sure the memory
          // required by configured read-only threads does not exceed the total system virtual memory.
-         auto get_total_vm_size = []() -> unsigned long {
+         auto get_vm_sizes = []() -> std::pair<size_t, size_t> {
             std::string attr_name;
+            size_t vm_total { 0 };
+            size_t vm_used { 0 };
             std::ifstream meminfo_file("/proc/meminfo");
             while (meminfo_file >> attr_name) {
                if (attr_name == "VmallocTotal:") {
-                  unsigned long vm_total;
-                  if (meminfo_file >> vm_total) {
-                     return vm_total;
-                  } else {
-                     return 0;
-                  }
+                  if ( !(meminfo_file >> vm_total) )
+                     break;
+               } else if (attr_name == "VmallocUsed:") {
+                  if ( !(meminfo_file >> vm_used) )
+                     break;
                }
                meminfo_file.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
             }
-            return 0;
+            return std::make_pair(vm_total, vm_used);
          };
 
-         unsigned long vm_total_kb = get_total_vm_size();
+         auto [vm_total_kb, vm_used_kb] = get_vm_sizes();
          EOS_ASSERT( vm_total_kb > 0, plugin_config_exception, "Unable to get system virtual memory size (not a Linux?), therefore cannot determine if the system has enough virtual memory for multi-threaded read-only transactions on EOS VM OC");
-         auto num_threads_supported = vm_total_kb / 4200000000;
-         // reserve 1 for the main thread
-         EOS_ASSERT( num_threads_supported > my->_ro_thread_pool_size + 1, plugin_config_exception, "--read-only-threads (${th}) greater than number of threads supported for EOS VM OC (${supp})", ("th", my->_ro_thread_pool_size) ("supp", num_threads_supported - 1 ) );
+         // reserve 1 for the app thread, 1 for anything else which might use VM
+         int num_threads_supported = (vm_total_kb - vm_used_kb) / 4200000000 - 2;
+         EOS_ASSERT( num_threads_supported >= my->_ro_thread_pool_size, plugin_config_exception, "--read-only-threads (${th}) greater than number of threads supported for EOS VM OC (${supp})", ("th", my->_ro_thread_pool_size) ("supp", num_threads_supported) );
```

Comment:
Would be nice to log `vm_total_kb` and `vm_used_kb` and calculation that went into determining `num_threads_supported`; either when it fails or always.

Reply:
Done.